bin/test.sh

```diff
@@ -60,18 +60,15 @@ BOOKINFO_DEPLOYMENTS="details-v1 productpage-v1 ratings-v1 reviews-v1 reviews-v2
 
 if [ "$SKIP_SETUP" -ne 1 ]; then
     kubectl create ns bookinfo || /bin/true
-    kubectl label ns bookinfo istio-env=istio-control --overwrite
-    #kubectl delete -f samples/bookinfo/platform/kube/bookinfo.yaml --ignore-not-found
-    #kubectl delete -f samples/bookinfo/networking/destination-rule-all-mtls.yaml --ignore-not-found
-    #kubectl delete -f samples/bookinfo/networking/bookinfo-gateway.yaml --ignore-not-found
+    ISTIO_CONTROL=$(kubectl get namespaces -o=jsonpath='{$.items[:1].metadata.labels.istio-env}' -l istio-env)
+    ISTIO_CONTROL=${ISTIO_CONTROL:-istio-control}
+
+    kubectl label ns bookinfo istio-env=${ISTIO_CONTROL} --overwrite
 
     kubectl -n bookinfo apply -f samples/bookinfo/platform/kube/bookinfo.yaml
```

Comment:
Would it be possible to switch this to a fortio deployment ? Bookinfo is super heavy ( java, etc), we can leave it in post-submit / stability tests.

Reply:
Using something more lightweight than Bookinfo is a good idea. Fortio has the following problems, though:
* Fortio doesn't support graceful shutdowns, i.e. it can send incomplete responses during shutdown. This is a problem when we have to restart pods while switching to a new control plane.
* Fortio starts serving HTTP-requests before the server is fully configured. This could result in 404s during startup.

Should we try to fix these issues in fortio or should we write a small HTTP server which handles startup and shutdown correctly?
Our target is to prove that an istio update can be done in an HA setup.